pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -170,6 +170,15 @@ func ResolveAuthRefAndPodIdentity(ctx context.Context, client client.Client, log
 			authParams["awsRoleArn"] = serviceAccount.Annotations[kedav1alpha1.PodIdentityAnnotationEKS]
 		} else if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAwsKiam {
 			authParams["awsRoleArn"] = podTemplateSpec.ObjectMeta.Annotations[kedav1alpha1.PodIdentityAnnotationKiam]
+		} else if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAzure {
+			// Check if podidentity is empty or default
+			if podIdentity.IdentityID == nil {
+				// Set to empty in case of trigger nil pointer error
+				podIdentity.IdentityID = new(string) 
+			} else if *podIdentity.IdentityID == "" {
+				return nil, kedav1alpha1.AuthPodIdentity{Provider: kedav1alpha1.PodIdentityProviderNone},
+					fmt.Errorf("error getting IdentityID, IdentityID cannot be empty")
```

Comment:
Should we raise an error here? I mean, isn't a warning enough? We could add a validation in validation as part of the admission webhooks if we want to prevent this situation